src/tools/synth_node.rs

```diff
@@ -26,22 +27,20 @@ pub struct SyntheticNode {
 }
 
 impl SyntheticNode {
-    pub async fn new(config: pea2pea::Config) -> Self {
+    pub async fn new(config: &TestConfig) -> Self {
         let (sender, receiver) = mpsc::channel(SYNTH_NODE_QUEUE_DEPTH);
         let inner = InnerNode::new(config, sender).await;
-        inner.enable_handshake().await;
+        if config.synth_node_config.do_handshake {
+            inner.enable_handshake().await;
+        }
         inner.enable_reading().await;
         inner.enable_writing().await;
         Self { inner, receiver }
     }
 
-    /// Starts node bound to localhost and with default [pea2pea::Config]
-    pub async fn start() -> Result<Self, AddrParseError> {
-        let node_config = pea2pea::Config {
-            listener_ip: Some("127.0.0.1".parse()?),
```

Comment:
is this property preserved?

Reply:
it is in `pea2pea`, but have you checked in this repo's tests? I have doubts :detective:

Reply:
Is this what you mean:
```
impl TestConfig {
    pub fn new() -> io::Result<Self> {
        Ok(Self {
            real_node_config: NodeConfig::new()?,
            synth_node_config: Default::default(),
            pea2pea_config: pea2pea::Config {
                listener_ip: Some("127.0.0.1".parse().unwrap()),
                ..Default::default()
            },
        })
    }

```